src/cpp/rtps/builtin/data/WriterProxyData.cpp

```diff
@@ -352,6 +351,76 @@ uint32_t WriterProxyData::get_serialized_size(
             type_information);
     }
 
+    if (dds::QosPoliciesSerializer<dds::HistoryQosPolicy>::should_be_sent(history))
+    {
+        ret_val += dds::QosPoliciesSerializer<dds::HistoryQosPolicy>::cdr_serialized_size(history.value());
+    }
+
+    // Check if serialize_optional_qos property is set to true
+    bool serialize_optional_qos = false;
+    if (properties.size() > 0)
+    {
+        auto serialize_opt_qos_property = std::find_if(
+            properties.begin(),
+            properties.end(),
+            [](const fastdds::dds::ParameterProperty_t& property)
+            {
+                return property.first() == fastdds::dds::parameter_serialize_optional_qos;
+            });
+
+        if (serialize_opt_qos_property != properties.end())
+        {
+            if (serialize_opt_qos_property->second() == "true")
+            {
+                serialize_optional_qos = true;
+            }
+        }
+    }
```

Comment:
As we internally agreed, we could move the property to be in the `participant` and leave a boolean `should_send_optional_qos` in the Reader/Writer proxy datas to be filled after the proxy datas creations.

We should also check for "1" value